### PR TITLE
structure: fix err check

### PR DESCRIPTION
**Pull request by Tjianke**
### What problem does this PR solve?

Issue Number: #22979  

Problem Summary:

### What is changed and how it works?

What's Changed: added err check

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release note

### CODE REVIEW

File: structure/structure_test.go [hunk 1]
```diff
@@ -53,7 +53,11 @@ func (s *testTxStructureSuite) TearDownSuite(c *C) {
 func (s *testTxStructureSuite) TestString(c *C) {
 	txn, err := s.store.Begin()
 	c.Assert(err, IsNil)
-	defer txn.Rollback()
+	defer func() {
+		// to fix: invalid txn
+		err = txn.Rollback()
```
Comment (zimulala) [inline structure/structure_test.go:58]: We commit this `txn` at the end of this function. So I think we can remove this code.

File: structure/structure_test.go [hunk 2]
```diff
@@ -102,7 +106,11 @@ func (s *testTxStructureSuite) TestString(c *C) {
 func (s *testTxStructureSuite) TestList(c *C) {
 	txn, err := s.store.Begin()
 	c.Assert(err, IsNil)
-	defer txn.Rollback()
+	defer func() {
+		err = txn.Rollback()
+		// to fix: test for invalid txn
```
Comment (zimulala) [inline structure/structure_test.go:111]: Please check whether we need to remove this code.

File: structure/structure_test.go [hunk 3]
```diff
@@ -210,7 +218,11 @@ func (s *testTxStructureSuite) TestList(c *C) {
 func (s *testTxStructureSuite) TestHash(c *C) {
 	txn, err := s.store.Begin()
 	c.Assert(err, IsNil)
-	defer txn.Rollback()
+	defer func() {
+		// to fix: invalid transaction
+		err := txn.Rollback()
```
Comment (zimulala) [inline structure/structure_test.go:223]: Ditto.

**Review by zz-jason — APPROVED** [review]
LGTM

**Pull request merged**